src/foundry/common/documents/setting.mjs.d.ts

```diff
@@ -0,0 +1,87 @@
+// FOUNDRY_VERSION: 10.291
+
+import Document, { DocumentMetadata } from "../abstract/document.mjs";
+import * as fields from "../data/fields.mjs";
+
+declare global {
+  type SettingData = BaseSetting.Properties;
+}
+
+/**
+ * The Document definition for a Setting.
+ * Defines the DataSchema and common behaviors for a Setting which are shared between both client and server.
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface BaseSetting extends BaseSetting.Properties {}
+declare class BaseSetting extends Document<BaseSetting.SchemaField, BaseSetting.Metadata> {
```

Comment:
This line gets the following error:
> Type 'SchemaField' does not satisfy the constraint 'AnyWithFlags'.
> Property 'flags' is missing in type 'Schema' but required in type '{ flags: ObjectField<{}, Record<string, unknown> | null | undefined, Record<string, unknown>, Record<string, unknown>>; }'.ts(2344)

The trouble is, the BaseSetting schema doesn't have a flags field.

![image](https://github.com/League-of-Foundry-Developers/foundry-vtt-types/assets/4632129/4468c27d-a1c4-4832-9a8e-7caf17980b93)

![image](https://github.com/League-of-Foundry-Developers/foundry-vtt-types/assets/4632129/9e34039d-d79a-46dd-8e3c-4564342be21c)

Reply:
Hmm. Foundry violating its own patterns again. This isn't the first time. Although I'm not entirely sure anymore why we said that flags are required. It might be that's because `Document` has methods to fetch flags. We will have to look whether we can just loosen that restriction.

Reply:
Sounds good to me. I will leave this with the type error for now.